src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/DataAnnotationsMetadataProvider.cs

```diff
@@ -80,23 +89,55 @@ public void CreateDisplayMetadata(DisplayMetadataProviderContext context)
                 displayMetadata.DataTypeName = DataType.Html.ToString();
             }
 
-            // Description
+            // DisplayFormatString
+            if (displayFormatAttribute != null)
+            {
+                displayMetadata.DisplayFormatString = displayFormatAttribute.DataFormatString;
+            }
+
+            // DisplayDescription
```

Comment:
Just `// Description` (that's the name of the property being set) and similar comments below. Also note "description" comes before `display" and there's no reason to reorder this code.